packages/@aws-cdk/aws-ec2/README.md

```diff
@@ -608,6 +608,32 @@ new ec2.Instance(this, 'Instance', {
 
 ```
 
+## Volumes
+
+Whereas Block Devices are created, and automatically associated, with a specific instance when that instance is created. Volumes allow you to create a block device that can be attached to, or detached from, any instance at any time. Some types of Volumes can also be attached to multiple instances at the same time to allow you to have shared storage between those instances. A notable restriction is that a Volume can only be attached to instances in the same availability zone.
+
+The following demonstrates how to create a 500 GiB encrypted Volume in the `us-west-2a` availability zone, and give a role the ability to attach that Volume to a specific instance:
+```ts
+const instance = new ec2.Instance(this, 'Instance', {
+  // ... 
```

Comment:
I imagine the instance still needs some magic UserData here to attach the volume, right?

Can you add an example of that?

Reply:
Yeah, it's pretty involved though. If I limit it to just attaching then it's much more reasonable. At creation, the EBS volume is not formatted, so after attaching it for the first time you'd typically have to format the device as well; but you'd only want to do that conditionally, in case it's not the first attachment.

How in-depth is reasonable for the readme?

Reply:
I added a fair bit to the README. Please let me know what you think.